<launch>
  <!--  Launch file for scene 3, feature related parameters are loaded later -->

  <!--  PARAMETERS -->
  <!-- scene file -->
  <param name="/simulator/scene_file" value="$(find simulator)scenarios/scene3.xml" type="string"/>
  <param name="/irl_features/update_type" value="oracle" type="string"/>
  <param name="/irl_features/sensor_range" value="30.0" type="double"/>
  
  <!-- grid related parameters -->
  <param name="/irl_features/goal" value="[55.0, 110.0, 0.0, 0.0]"/>
  <param name="/irl_features/x1" value="0.0" type="double"/>
  <param name="/irl_features/y1" value="0.0" type="double"/>
  <param name="/irl_features/x2" value="130.0" type="double"/>
  <param name="/irl_features/y2" value="130.0" type="double"/>

  <!-- other parameters -->
  <param name="/irl_features/cell_size" value="1.0" type="double"/>
  <param name="/irl_features/speed" value="1.0" type="double"/>
  <param name="/irl_features/goal_threshold" value="2.0" type="double"/>
  <param name="/irl_features/target_type" value="2" type="int"/>
  <param name="/irl_features/max_msg_age" value="0.3" type="double"/>

  <!-- robot control parameters -->
  <param name="/irl_features/move_robot" value="400" type="double"/>
  <param name="/irl_features/teleop_state" value="0.0" type="double"/>


  <!-- NODES -->
  <!-- main simulator node -->
  <node name="simulator" pkg="simulator" type="simulator" output="log"/>

  <!-- tf broadcaster for aligning frames -->
  <node name="simulator_tf_broadcaster" pkg="simulator" type="dummy_transforms.py" respawn="false" output="log" />

  <!-- feature logging -->
  <node name="feature_extraction" pkg="feature_extraction" type="feature_extraction" respawn="false" output="screen" />

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find simulator)/rviz/scene3.rviz"/>

  <!-- planning node -->
  <node name="irl_features" pkg="momo" type="momo_ros.py" respawn="false" output="log"/>
</launch>
